app/models/search/work_search_form.rb

```diff
@@ -247,4 +221,28 @@ def default_sort_direction
     end
   end
 
+  ###############
+  # COUNTING
+  ###############
+
+  def self.count_for_user(user)
+    Rails.cache.fetch(count_cache_key(user), count_cache_options) do
+      WorkQuery.new(user_ids: [user.id]).count
+    end
+  end
+
+  def self.count_for_pseud(pseud)
+    Rails.cache.fetch(count_cache_key(pseud), count_cache_options) do
+      WorkQuery.new(pseud_ids: [pseud.id]).count
+    end
+  end
+
+  def self.count_cache_key(owner)
+    status = User.current_user ? 'logged_in' : 'logged_out'
+    "work_count_#{owner.class.name.underscore}_#{owner.id}_#{status}"
```

Comment:
To have the old cache invalidation behavior we just use `#{owner.works_index_cache_key}` instead of `#{owner.class.name.underscore}_#{owner.id}`.

Reply:
1. Should we add a comment saying that's an option available to us, in case we decide to revisit this?
2. Just checking my understanding -- `owner.class.name.underscore` the `underscore` is just a precaution and right now it could only be `user` or `pseud`?

Reply:
1. I added a comment.
2. For `owner.class.name.underscore`, I just followed `lib/works_owner.rb`'s example. For this method, it will only be `user` or `pseud` (a "work owner" could be `tag` or `collection` but we don't pass them in here).